core/jdiameter/impl/src/main/java/org/jdiameter/client/impl/controller/PeerImpl.java

```diff
@@ -1060,10 +1060,9 @@ public boolean receiveMessage(IMessage message) {
           if (isRedirectAnswer(avpResCode, message)) {
             message.setListener(request.getEventListener());
             message = processRedirectAnswer(request, message);
-            //if return value is not null, there was some error, lets try to invoke listener if it exists...
+            // if return value is not null, there was some error, lets try to invoke listener if it exists...
             isProcessed = message == null;
           }
-          avpResCode = message.getAvps().getAvp(RESULT_CODE);

```

Comment:
I'm wondering now whether, because if a redirect was processed, the message (i.e. Answer) could possibly be a different message to when avpResCode was originally assigned its value, this assignment statement should be added again, to ensure the avpResCode contains the most recent value, and this particular change reverted?

Reply:
Reverted